api/src/main/java/org/pmiops/workbench/cohortbuilder/FieldSetQueryBuilder.java

```diff
@@ -0,0 +1,303 @@
+    }
+    ColumnConfig columnConfig = tableQueryAndConfig.getColumn(columnFilter.getColumnName());
+    if (columnConfig == null) {
+      throw new BadRequestException("Column " + columnFilter.getColumnName() + " does not exist");
+    }
+    Operator operator = columnFilter.getOperator();
+    if (operator == null) {
+      operator = Operator.EQUAL;
+    }
+
+    if (operator.equals(Operator.IN)) {
+      handleInClause(columnFilter, columnConfig, sqlBuilder, paramMap);
+    } else {
+      handleComparison(columnFilter, columnConfig, sqlBuilder, paramMap);
+    }
+  }
+
+  private void handleColumnFilters(List<ColumnFilter> columnFilters,
+      TableQueryAndConfig tableQueryAndConfig,
+      StringBuilder sqlBuilder, Map<String, QueryParameterValue> paramMap) {
+    boolean first = false;
+    if (columnFilters.isEmpty()) {
+      throw new BadRequestException("Empty column filter list is invalid");
+    }
+    if (columnFilters.size() == 1) {
+      handleColumnFilter(columnFilters.get(0), tableQueryAndConfig, sqlBuilder, paramMap);
+    } else {
+      sqlBuilder.append("(");
+      for (ColumnFilter columnFilter : columnFilters) {
+        if (first) {
+          first = false;
+        } else {
+          sqlBuilder.append("\nand\n");
+        }
+        handleColumnFilter(columnFilter, tableQueryAndConfig, sqlBuilder, paramMap);
+      }
+      sqlBuilder.append(")");
+    }
+  }
+
+  public QueryJobConfiguration buildQuery(ParticipantCriteria participantCriteria,
+      TableQueryAndConfig tableQueryAndConfig, long resultSize, long offset) {
+    TableQuery tableQuery = tableQueryAndConfig.getTableQuery();
+    TableConfig tableConfig = tableQueryAndConfig.getTableConfig();
+    List<String> columnNames = tableQuery.getColumns();
+    String tableName = tableQuery.getTableName();
+
+    StringBuilder startSql = new StringBuilder("select ");
+    startSql.append(Joiner.on(", ").join(columnNames));
+    startSql.append("\nfrom `${projectId}.${dataSetId}.");
+    startSql.append(tableName);
+    startSql.append("` ");
+    startSql.append(tableName);
+    startSql.append("\nwhere\n");
+
+    Map<String, QueryParameterValue> paramMap = new HashMap<>();
+    List<List<ColumnFilter>> columnFilters = tableQuery.getFilters();
+    if (columnFilters != null && !columnFilters.isEmpty()) {
+      if (columnFilters.size() == 1) {
+        handleColumnFilters(columnFilters.get(0), tableQueryAndConfig, startSql, paramMap);
+      } else {
+        startSql.append("(");
+        boolean first = true;
+        for (List<ColumnFilter> filterList : columnFilters) {
+          if (first) {
+            first = false;
+          } else {
+            startSql.append("\nor\n");
+          }
+          handleColumnFilters(filterList, tableQueryAndConfig, startSql, paramMap);
+        }
+        startSql.append(")");
+      }
+      startSql.append("\nand\n");
+    }
+
+    StringBuilder endSql = new StringBuilder("order by ");
```

Comment:
This assumes orderBy.size() > 0; would be better to explicitly validate this. (I realize that materializeCohort happens to satisfy this precondition).